api/v1alpha1/providertemplate_types.go

```diff
@@ -15,104 +15,64 @@
 package v1alpha1
 
 import (
-	"fmt"
-
-	"github.com/Masterminds/semver/v3"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	// ChartAnnotationCAPIVersion is an annotation containing the CAPI exact version in the SemVer format associated with a ProviderTemplate.
-	ChartAnnotationCAPIVersion = "hmc.mirantis.com/capi-version"
-	// ChartAnnotationCAPIVersionConstraint is an annotation containing the CAPI version constraint in the SemVer format associated with a ProviderTemplate.
-	ChartAnnotationCAPIVersionConstraint = "hmc.mirantis.com/capi-version-constraint"
+	// ChartAnnotationCAPIContractVersion is an annotation containing the expected core CAPI contract version (e.g. v1beta1) associated with a ProviderTemplate.
+	ChartAnnotationCAPIContractVersion = "hmc.mirantis.com/capi-version"
 )
 
-// +kubebuilder:validation:XValidation:rule="!(has(self.capiVersion) && has(self.capiVersionConstraint))", message="Either capiVersion or capiVersionConstraint may be set, but not both"
-
 // ProviderTemplateSpec defines the desired state of ProviderTemplate
 type ProviderTemplateSpec struct {
 	Helm HelmSpec `json:"helm,omitempty"`
-	// CAPI exact version in the SemVer format.
-	// Applicable only for the cluster-api ProviderTemplate itself.
-	CAPIVersion string `json:"capiVersion,omitempty"`
-	// CAPI version constraint in the SemVer format indicating compatibility with the core CAPI.
-	// Not applicable for the cluster-api ProviderTemplate.
-	CAPIVersionConstraint string `json:"capiVersionConstraint,omitempty"`
-	// Providers represent exposed CAPI providers with exact compatibility versions set.
+	// CAPI [contract version] indicating compatibility with the core CAPI.
+	// Currently supported versions: v1alpha3_v1alpha4_v1beta1.
+	// The field is not applicable for the cluster-api ProviderTemplate.
```

Comment:
I think it is now. For core capi - provided contracts, for other - expected contracts

Reply:
I think the comment is related to the two questions regarding either hardcoding the CAPI contract versions or fetching them during the validation ([one](https://github.com/Mirantis/hmc/blob/b03b9b3767fe281aeac35a58431aec67134c225e/internal/webhook/management_webhook.go#L83), [two](https://github.com/Mirantis/hmc/blob/b03b9b3767fe281aeac35a58431aec67134c225e/internal/controller/management_controller.go#L429-L432))

At the moment I do not think this might be a good approach to have a single with for two similar but not the same purposes